src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -112,7 +112,9 @@ bool SHAMap::walkBranch (SHAMapTreeNode* node,
     return true;
 }
 
-bool SHAMap::compare (SHAMap::ref otherMap, Delta& differences, int maxCount)
+bool
+SHAMap::compare (std::shared_ptr<SHAMap> const& otherMap,
+                 Delta& differences, int maxCount)
```

Comment:
I can dig back into this.  However I strongly recommend I do so under another pull request.  This P/R is very big, but very simple and mechanical.  Making `compare const` will be more complex, though hopefully small.  And should be isolated so that it can be more easily analyzed for correctness by reviewers.